### DISCUSSION TITLE
Crackling Audio when paused or no input.

### DISCUSSION THREAD

**Discussion by mvoss96** [pschatzmann/arduino-snapclient#40] [Q&A] [answered]
Hi, 
thanks again for your great work.
Im trying out the SnapClientFreeRTOS example modified to i2s output.

After increasing the buffers everything works fine, however when the audio input is paused or stopped i get a crackling sound instead of silence. Any Idea why that might be and how i could prevent it?

Also i get the same issue when i change the volume in the snapcast webserver gui. The audio also starts to crackle

My modified example code:

```
#include "AudioTools.h"
#include "SnapClient.h"
#include "api/SnapProcessorRTOS.h"  // install https://github.com/pschatzmann/arduino-freertos-addons
#include "AudioCodecs/CodecOpus.h"  // https://github.com/pschatzmann/arduino-libopus


I2SStream out;
OpusAudioDecoder opus;
WiFiClient wifi;
SnapProcessorRTOS rtos(1024 * 64);  // define queue with 8 kbytes
SnapTimeSyncDynamic synch(172, 10);  // optional configuratioin
SnapClient client(wifi, out, opus);

#define I2S_DOUT 6
#define I2S_BCLK 4
#define I2S_LRC 5

const char *ssid = "";
const char *password = "";

void setup() {
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Info);

  // login to wifi -> Define values in SnapConfig.h or replace them here
  WiFi.begin(ssid, password);
  Serial.print("Connecting to WiFi ..");
  while (WiFi.status() != WL_CONNECTED) {
    Serial.print('.');
    delay(1000);
  }

  // initialize output
  auto cfg = out.defaultConfig();
  cfg.pin_ws = I2S_LRC;
  cfg.pin_bck = I2S_BCLK;
  cfg.pin_data = I2S_DOUT;
  cfg.buffer_size = 1024;
  cfg.buffer_count = 50;
  out.begin(cfg);

  // print ip address
  Serial.println();
  Serial.println(WiFi.localIP());

  // Use FreeRTOS
  client.setSnapProcessor(rtos);

  // Define CONFIG_SNAPCAST_SERVER_HOST in SnapConfig.h or here
  client.setServerIP(IPAddress(192, 168, 2, 30));


  // start snap client
  client.begin(synch);
}


void loop() {
  bool res = client.doLoop();
  if (!res) {
    Serial.println("Ha");
  }
}

```

**Comment by pschatzmann** [comment, marked as answer]
Are you sure that you use the latest version of all libraries.
With the new I2S driver, I have reintroduced the [auto_clear parameter](https://github.com/pschatzmann/arduino-audio-tools/commit/2f35a96370486abe3e5e06a7a4156f5b65aabbca).

Maybe this is also a feature of your DAC ? 
What happens if you mute ?

**Reply by mvoss96** [reply]
Thanks for your reply,
I updated the audio-tools lib to version 1.0 now the crackling in pauses is gone.
However i still get crackling when i change the volume in the snapcast server web gui

**Comment by speciouz** [comment]
Thank you for your hard work!

I was having the same issue, but since I'm using SPDIF output, i2s_auto_clear did not work for me. However, I managed to find a workaround.

My setup is as follows:
ESP32 -> Digital to Analog Converter -> TDA2030 -> Speaker

I turned down the TDA2030 module's volume and tried to set the volume on the software side like this:
`client.setVolumeFactor(4);`

Even though I received an error:
`[E] VolumeStream.h : 192 - Invalid volume: 4.000000`
it "kinda" worked.

Thank you again.
